Mathlib/Logic/Equiv/Basic.lean

```diff
@@ -0,0 +1,1811 @@
+/-
+Copyright (c) 2015 Microsoft Corporation. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Leonardo de Moura, Mario Carneiro
+-/
```

Comment:
If you'd like, you can add a 
```
Ported by: <your name>
```
here. We've been meaning to start doing this to properly credit people!